subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -687,14 +687,21 @@ private synchronized void initAllArtifacts() {
     }
 
     public Set<ExcludeRule> getExcludeRules() {
+        initExcludeRules();
+        return Collections.unmodifiableSet(parsedExcludeRules);
+    }
+
+    /**
+     * Synchronize read access to excludes. Mutation does not need to be thread-safe.
+     */
+    private synchronized void initExcludeRules() {
```

Comment:
This is likely the wrong solution. 

The per project model is inherently not thread safe. Playing whack-a-mole in attempt to feign thread safety won't work. Access serialization should be more coarse grained.

Reply:
Understood, will discuss with the dependency management team. But since this fix is on master already and its absence causes flaky tests, I will still merge this for now.